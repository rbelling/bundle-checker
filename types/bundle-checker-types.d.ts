export interface IBundleCheckerReport {
  reportText: string;
}

export interface IBundleCheckerParams {
  // The build script that will be run as part of the bundle checker.
  buildScript: string;
  // Optional distPath represents the dist folder. If present, all of the targetFilesPattern globs will be prepended with this
  distPath: string;
  // The install script that downloads and install dependencies before the build can start.
  installScript: string;
  // Byte-size limit of parsed code (non-gzipped)
  sizeLimit?: number;
  // An array of patterns that is used to target distribution files.
  targetFilesPattern: string[];
  // Github repo to clone and analyze
  gitRepository: string;
  currentBranch: string;
  targetBranch: string;
}

export interface ITotalSize {
  js: number;
  css: number;
}
