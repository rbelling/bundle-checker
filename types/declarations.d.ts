declare module 'size-limit' {
  interface ISizeLimitResponse {
    gzipped: number;
    parsed: number;
  }
  export default function(files: string[], options?: any): Promise<ISizeLimitResponse>;
}
