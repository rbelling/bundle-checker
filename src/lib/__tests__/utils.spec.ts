import { ITableReport, ITableRow } from '../../../types/bundle-checker-types';
import {
  createMarkdownTable,
  getRowsForTotalSizeReport,
  groupByFileExtension,
  withDeltaSize
} from '../utils';
describe('generating markdown tables', () => {
  it('build a markdown table with the content given', () => {
    const headers = ['git branch', 'file size'] as ITableRow;
    const rows = [
      ['master️', 'js: 356.6kB, css: 0'],
      ['develop', 'js: 376.4kB, css: 0']
    ] as ITableRow[];

    const table = createMarkdownTable([headers, ...rows]);

    expect(table).toMatchSnapshot();
  });

  it('Groups targeted files by extension', () => {
    const targetedFiles = [
      'main.js',
      'main.css',
      'shared.js',
      'shared.css',
      'sprite.png',
      'font.woff',
      'font-italic.123716.woff'
    ];
    const expectedGrouping = {
      css: ['main.css', 'shared.css'],
      js: ['main.js', 'shared.js'],
      png: ['sprite.png'],
      woff: ['font.woff', 'font-italic.123716.woff']
    };

    expect(groupByFileExtension(targetedFiles)).toMatchObject(expectedGrouping);
  });

  it(`Prints the delta size with a triangle arrow`, () => {
    expect(withDeltaSize(1024, 2048)).toBe(`2KB (🔺 +1KB)`);
    expect(withDeltaSize(2048, 512)).toBe(`512B (▼ -1.5KB)`);
  });

  it('Creates rows for total size report in the expected format', () => {
    const targetBranchReport: ITableReport = {
      css: 150,
      js: 1000
    };
    const currentBranchReport: ITableReport = {
      jpg: 2000,
      js: 1100
    };
    const expectedFormat: ITableRow[] = [
      ['.css', '150B', '0B (▼ -150B)'],
      ['.jpg', '0B', '1.95KB (🔺 +1.95KB)'],
      ['.js', '1000B', '1.07KB (🔺 +100B)']
    ];

    expect(getRowsForTotalSizeReport(targetBranchReport, currentBranchReport)).toEqual(
      expectedFormat
    );
  });
});
