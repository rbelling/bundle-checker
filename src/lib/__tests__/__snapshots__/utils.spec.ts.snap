// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generating markdown tables build a markdown table with the content given 1`] = `
"| git branch | file size |
| --- | --- |
| masterÔ∏è | js: 356.6kB, css: 0 |
| develop | js: 376.4kB, css: 0 |"
`;
